aspnetcore/fundamentals/openapi/buildtime-openapi.md

```diff
@@ -36,65 +37,99 @@ Run the following command in the directory that contains the project file:
+### Modify the output directory of the generated Open API file
 
-```xml
-<PropertyGroup>
-  <OpenApiGenerateDocumentsOptions>--file-name my-open-api</OpenApiGenerateDocumentsOptions>
-</PropertyGroup>
-```
+By default, the generated OpenAPI document is generated in the `obj` directory. The value of the `OpenApiDocumentsDirectory` property:
 
-### Selecting the OpenAPI document to generate
+* Sets the location of the generated file.
+* Is resolved relative to the project file.
 
-Some applications may be configured to emit multiple OpenAPI documents, for various versions of an API or to distinguish between public and internal APIs. By default, the build-time document generator will emit files for all documents that are configured in an application. To only emit for a single document name, set the `--document-name` argument in the `OpenApiGenerateDocumentsOptions` property.
+The following example generates the OpenAPI document in the same directory as the project file:
 
-```xml
-<PropertyGroup>
-  <OpenApiGenerateDocumentsOptions>--document-name v2</OpenApiGenerateDocumentsOptions>
-</PropertyGroup>
-```
+:::code language="xml" source="~/fundamentals/openapi/samples/9.x/BuildTime/csproj/MyTestApi.csproj.php" id="snippet_1" highlight="2":::
 
-## Customizing run-time behavior during build-time document generation
+In the following example, the OpenAPI document is generated in the `MyOpenApiDocs` directory, which is a sibling directory to the project directory:
 
-Under the hood, build-time OpenAPI document generation functions by launching the application's entrypoint with an inert server implementation. This is a requirement to produce accurate OpenAPI documents since all information in the OpenAPI document cannot be statically analyzed. Because the application's entrypoint is invoked, any logic in the applications' startup will be invoked. This includes code that injects services into the DI container or reads from configuration. In some scenarios, it's necessary to restrict the codepaths that will run when the application's entry point is being invoked from build-time document generation. These scenarios include:
+:::code language="xml" source="~/fundamentals/openapi/samples/9.x/BuildTime/csproj/MyTestApi.csproj.php" id="snippet2" highlight="2":::
 
-- Not reading from certain configuration strings
-- Not registering database-related services
+### Modify the output file name
 
-In order to restrict these codepaths from being invoked by the build-time generation pipeline, they can be conditioned behind a check of the entry assembly like so:
+By default, the generated OpenAPI document has the same name as the app's project file. To modify the name of the generated file, set the `--file-name` argument in the `OpenApiGenerateDocumentsOptions` property:
```

Comment:
This isn't what I've seen in my testing. I get something like `ProjectName_DocumentName.json` (e.g. `API_v1.json`).

Reply:
> This isn't what I've seen in my testing. I get something like `ProjectName_DocumentName.json` (e.g. `API_v1.json`).

My build is 9.0.100-rc.1.24415.1
run /TestScripts/create.cmd